adapter/internal/discovery/xds/server.go

```diff
@@ -882,129 +930,129 @@ func UpdateEnforcerThrottleData(throttleData *throttle.ThrottleData) {
-		Version:                version,
-	}
-	SetSuccessState(label, enforcerNewState, routerNewState)
-}
-
-// restorePreviousState retrive the last successful state from the state cache and restore the date to the maps.
-func restorePreviousState(label string) string {
-	// Get the last known successful state of enforcer and router from the state cache
-	enforcerOldState, routerOldState := GetLastSuccessState(label)
-	// Initialize new maps and populate the success state from the retrieved state cache.
-
-	// Enforcer resources
-	apiMgwSwaggerMap = enforcerOldState.Apis
-	openAPIEnforcerApisMap = enforcerOldState.OpenAPIEnforcerApisMap
-	apiToVhostsMap = enforcerOldState.APIToVhostsMap
-	apiUUIDToGatewayToVhosts = enforcerOldState.APIUUIDToGatewayToVhosts
-
-	// Router resources
-	openAPIRoutesMap = routerOldState.APIRoutesMap
-	openAPIClustersMap = routerOldState.APIClustersMap
-	openAPIEndpointsMap = routerOldState.APIEndpointsMap
-	envoyListenerConfigMap = routerOldState.EnvoyListenerConfigMap
-	envoyRouteConfigMap = routerOldState.EnvoyRouteConfigMap
-	return enforcerOldState.Version
-}
-
-// TODO: (VirajSalaka) Remove after the fallback mechanism implementation, if it remains unused.
-/** This method will listen to the callback messages from the xds protocol.
-* A message will be published for each xds client request (for API resources only)
-* Based on the message (error/ success) the current state of the resources will be changed.
-* If failure ==> restore the last success state
-* If success ==> store the current state as the last success state
-* Flow -->
-* 1. Update the enforcer
-* 2. Listen for the discovery request from client.
-* 3. If success, set the current state as the success state. Otherwise, restore the previous state.
-* 4. Update the router
-**/
-func watchEnforcerResponse() {
-	for {
-		requestEvent := <-GetRequestEventChannel()
-		logger.LoggerXds.Debugf("xds Request from client version : %s", requestEvent.Version)
-		if requestEvent.IsError {
-			logger.LoggerXds.Infof("Applying config failed. Last success version of enforcer : %s", requestEvent.Version)
-			lastSuccessVersion := restorePreviousState(requestEvent.Node)
-			_, _, _, _, apis := GenerateEnvoyResoucesForLabel(requestEvent.Node)
-			UpdateEnforcerApis(requestEvent.Node, apis, lastSuccessVersion)
-		} else {
-			logger.LoggerXds.Infof("Successfully updated APIs in Client for version: %s", requestEvent.Version)
-
-			// Successful message, set the current state of the enforcer apis to the state cache.
-			buildAndStoreSuccessState(requestEvent.Node, requestEvent.Version)
-
-			// Generate the router resources and update the router.
-			listeners, clusters, routes, endpoints, _ := GenerateEnvoyResoucesForLabel(requestEvent.Node)
-			UpdateXdsCacheWithLock(requestEvent.Node, endpoints, clusters, routes, listeners)
-		}
-	}
-}
+// // buildAndStoreSuccessState method builds the latest successful enforcer and router resources and returns as
```

Comment:
why commented?

Reply:
This code is not being executed. Will remove this